api/src/main/java/org/openmrs/DrugOrder.java

```diff
@@ -512,13 +517,41 @@ public void setDosing(DosingInstructions di) {
 	 */
 	@Override
 	public boolean hasSameOrderableAs(Order otherOrder) {
-		if (!super.hasSameOrderableAs(otherOrder)) {
-			return false;
-		}
-		if (!(otherOrder instanceof DrugOrder)) {
-			return false;
-		}
-		DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
-		return OpenmrsUtil.nullSafeEquals(this.getDrug(), otherDrugOrder.getDrug());
+        if (!super.hasSameOrderableAs(otherOrder)) {
+            return false;
+        }
+        if (!(otherOrder instanceof DrugOrder)) {
+            return false;
+        }
+        DrugOrder otherDrugOrder = (DrugOrder) otherOrder;
+
+        if (isNonCodedDrug() || otherDrugOrder.isNonCodedDrug()) {
```

Comment:
For example, DrugOrderA is nonCoded but with a blank nonCodedValue, DrugOrderB is a regular drug order meaning it has a blank nonCodedValue too, it means we will get into this if clause and the comparison will return true when actually these are not the same based on our acceptance criteria on the ticket but if you use AND instead of OR then the if clause would fail and the logic stays correct. Of course the argument would be that DrugOrderA is technically invalid and should have already been caught by the drug order validator but the validators can be compromised in some cases depending on where you call the service method from which i won't elaborate how, in summary to be on a safe side, i would use AND instead of OR, makes sense?